tango_simlib/tango_sim_generator.py

```diff
@@ -211,17 +211,33 @@ def _reset_to_default_state(self):
             """Reset the model's quantities' adjustable attributes to their default
             values.
             """
+            type_map = {
+                'DevString': "",
+                'DevFloat': 0.0,
+                'DevDouble': 0.0,
+                'DevBoolean': False,
+                'DevEnum': 0,
+                'DevLong': 0,
+                'DevULong': 0,
+                'DevVoid': None}
             simulated_quantities = self.model.sim_quantities.values()
             for simulated_quantity in simulated_quantities:
                 sim_quantity_meta_info = simulated_quantity.meta
-                adjustable_attrs = simulated_quantity.adjustable_attributes
-
-                for attr in adjustable_attrs:
+                if 'initial_value' in sim_quantity_meta_info.keys():
+                    pass
```

Comment:
I'm not sure what it is happening here.  Does having an `initial_value` somehow result in `last_val` also getting set to that initial value?
Suggest inverting the logic, to simplify:
```
 if 'initial_value' not in sim_quantity_meta_info.keys():
    try:
        adjustable_val = ...
```

Reply:
The keys `initial_value` and `value` might be a problem if we handle them in the tango_sim_generator. In instances were we use the fgo and the simdd files, the `Quantity` object created will have both keys in its meta dictionary and that will create a confusion here as to which value to use to re-initialize the model's quantity values.

I would suggest we  handle this in the `PopulateModelQuantities` class. We create a new key, say call it `default_value` and check if the parsed info has the key `initial_value` or `value` and update the `default_value` accordingly.

Then in the `_reset_to_default_state` method we just look for the key `default_value` and update the `last_val` accordingly.

This is some of the output I got when using the fgo and simdd files with an attribute with initial values specified in both files:
SIMDD
```
{
   "dynamicAttributes": [
      {
          "basicAttributeData": {
             "name": "spfB5DefaultRfe2TempSetPoint",
             "dataSimulationParameters": {
                "quantity_simulation_type": "ConstantQuantity",
             }
          }
      }
   ]
}

```
Output of the `model.sim_quantities['spfB5DefaultRfe2TempSetPoint'].meta`:
```
{'alarms': {'delta_t': 'Not specified',
  'delta_val': 'Not specified',
  'extensions': '[]',
  'max_alarm': 'Not specified',
  'max_warning': 'Not specified',
  'min_alarm': 'Not specified',
  'min_warning': 'Not specified'},
 'color': 'Lime',
 'data_format': tango._tango.AttrDataFormat.SCALAR,
 'data_type': tango._tango.CmdArgType.DevEnum,
 'database': 'ska-lmc-vm:10000',
 'description': ' Default RFE 2nd stage temperature control setpoint ',
 'device': 'mid_dish_0001/elt/master',
 'display_unit': 'No display unit',
 'enum_labels': ['OFF', 'SP1', 'SP2', 'SP3', 'SP4', 'SP5', 'SP6'],
 'events': {'arch_event': {'archive_abs_change': 'Not specified',
   'archive_period': '60000',
   'archive_rel_change': 'Not specified',
   'extensions': '[]'},
  'ch_event': {'abs_change': 'Not specified',
   'extensions': '[]',
   'rel_change': 'Not specified'},
  'per_event': {'extensions': '[]', 'period': '60000'}},
 'format': '%s',
 'initial_value': '5',
 'label': 'b5DefaultRfe2TempSetPoint',
 'max_alarm': 'Not specified',
 'max_dim_x': 1,
 'max_dim_y': 0,
 'max_value': 'Not specified',
 'min_alarm': 'Not specified',
 'min_value': 'Not specified',
 'model': 'ska-lmc-vm:10000/mid_dish_0001/elt/master/spfB5DefaultRfe2TempSetPoint',
 'name': 'spfB5DefaultRfe2TempSetPoint',
 'polling': 0,
 'quality': 'ATTR_VALID',
 'quantity_simulation_type': 'ConstantQuantity',
 'standard_unit': 'No standard unit',
 'string': '4',
 'time': 1526654259.210451,
 'unit': '',
 'value': 4,
 'writable': 'READ_WRITE'}

```

Reply:
@kmadisa I agree, having the parsers standardise the model quantities so that there is always a `default_value` key-value pair sounds good.